### PR TITLE
Add index release sorter

### PR DESCRIPTION
**Pull request by fgimenez**

### CODE REVIEW

File: index_releases_sort.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package versionbundle
+
+import (
+	"github.com/coreos/go-semver/semver"
+)
+
+type SortIndexReleasesByVersion []Release
```
Comment (tuommaki) [inline index_releases_sort.go]: I think this should be `[]IndexRelease`.

File: index_releases_sort_test.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+import (
+	"reflect"
+	"sort"
+	"testing"
+)
+
+func TestIndexReleasesSortByVersion(t *testing.T) {
+	testCases := []struct {
+		name          string
+		releases      []Release
```
Comment (tuommaki) [inline index_releases_sort_test.go]: `s/Release/IndexRelease/g` :)

File: index_releases_sort.go (hunk 1 above)
Comment (kopiczko) [reply index_releases_sort.go]: According to go stdlib `ReleaseSlice`: https://golang.org/pkg/sort/#IntSlice

But maybe there can be different approaches to sorting, then this generic name doesn't work.

File: index_releases_sort_test.go (hunk 2 above)
Comment (fgimenez) [reply index_releases_sort_test.go]: Yep, just changed it in one place, should be done now PTAL.

File: index_releases_sort.go (hunk 1 above)
Comment (fgimenez) [reply index_releases_sort.go]: Indeed, changed to the proper type.

**Pull request merged**